src/Features/CSharp/Portable/MakeFieldReadonly/CSharpMakeFieldReadonlyDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.MakeFieldReadonly;
+
+namespace Microsoft.CodeAnalysis.CSharp.MakeFieldReadonly
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal class CSharpMakeFieldReadonlyDiagnosticAnalyzer :
+        AbstractMakeFieldReadonlyDiagnosticAnalyzer<ConstructorDeclarationSyntax, LambdaExpressionSyntax>
+    {
+        protected override void InitializeWorker(AnalysisContext context)
+            => context.RegisterSyntaxNodeAction(AnalyzeType, SyntaxKind.ClassDeclaration, SyntaxKind.StructDeclaration);
+
+        internal override bool CanBeReadonly(SemanticModel model, SyntaxNode node)
+        {
+            if (node.Parent is AssignmentExpressionSyntax assignmentNode && assignmentNode.Left == node)
```

Comment:
We have a helper called IsWrittenTo that you can use for this purpose.  It should replace this entire method.